ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/individual-journey/individual-journey.ts

```diff
@@ -39,15 +38,26 @@ export class IndividualJourney extends JourneyBase {
       return;
     }
 
+    // filter out completed hearings
+    const pending = upcoming.filter((answers: IndividualSuitabilityModel) =>
+      answers.selfTest === undefined || !answers.selfTest.isCompleted());
+
+    if (pending.length === 0) {
+      this.isDone = true;
+      return;
+    }
+
     // sort upcoming on date and pick the earliest
-    upcoming.sort((u1, u2) => u1.hearing.scheduleDateTime.getTime() - u2.hearing.scheduleDateTime.getTime());
-    this.currentModel = upcoming[0];
+    pending.sort((u1, u2) => u1.hearing.scheduleDateTime.getTime() - u2.hearing.scheduleDateTime.getTime());
+    this.currentModel = pending[0];
   }
 
   startAt(step: JourneyStep) {
     this.assertInitialised();
     if (this.isDone) {
       this.goto(IndividualJourneySteps.GotoVideoApp);
+    } else if (this.isQuestionnaireCompleted() && !this.isSelfTestStep(step)) {
+      this.goto(SelfTestJourneySteps.SameComputer);
     } else {
       this.goto(step);
```

Comment:
Missing case:
this.isQuestionnaireCompleted() && "Is a selftest step"

i.e. when a self test step is pasted in to browser, it will go there instead of going to first self test step

Reply:
I'm allowing you to go straight into a step because if you refresh the page that's what will happen and we want to allow that. But it's a good point, made me think :)